pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -136,11 +136,53 @@ func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev
 	if podSpec == nil {
 		podSpec = &corev1.PodSpec{}
 	}
+
 	if componentConfig == nil || componentConfig.NodePlacement == nil {
-		componentConfig = &v1.ComponentConfig{
-			NodePlacement: &v1.NodePlacement{},
+		switch nodePlacementOption {
+		case AnyNode:
+			componentConfig = &v1.ComponentConfig{NodePlacement: &v1.NodePlacement{}}
+
+		case RequireControlPlanePreferNonWorker:
+			componentConfig = &v1.ComponentConfig{
+				NodePlacement: &v1.NodePlacement{
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+								NodeSelectorTerms: []corev1.NodeSelectorTerm{
+									{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "node-role.kubernetes.io/control-plane",
+												Operator: corev1.NodeSelectorOpExists,
+											},
+										},
+									},
+								},
+							},
+							PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+								{
+									Weight: 100,
```

Comment:
Just to understand, is there a chance for this rule to conflict with another with similar weight?

Reply:
There is no another rule :)
This is only being set by default if the user did not specify any node placements.
I think that in such cases the weight is somewhat redundant.